flake.nix

```diff
@@ -0,0 +1,137 @@
+{
+  inputs = {
+    nixpkgs_.url = "github:deemp/flakes?dir=source-flake/nixpkgs";
+    nixpkgs.follows = "nixpkgs_/nixpkgs";
+    my-codium.url = "github:deemp/flakes?dir=codium";
+    drv-tools.url = "github:deemp/flakes?dir=drv-tools";
+    flake-utils_.url = "github:deemp/flakes?dir=source-flake/flake-utils";
+    flake-utils.follows = "flake-utils_/flake-utils";
+    haskell-tools.url = "github:deemp/flakes?dir=language-tools/haskell";
+    devshell.url = "github:deemp/flakes?dir=devshell";
+    my-lima.url = "github:deemp/flakes?dir=lima";
+  };
+  outputs =
+    { self
+    , flake-utils
+    , nixpkgs
+    , my-codium
+    , drv-tools
+    , haskell-tools
+    , devshell
+    , my-lima
+    , ...
+    }:
+    flake-utils.lib.eachDefaultSystem (system:
+    let
+      # We're going to make some dev tools for our Haskell package
+      # See NixOS wiki for more info - https://nixos.wiki/wiki/Haskell
+
+      # First, we import stuff
+      pkgs = nixpkgs.legacyPackages.${system};
+      inherit (my-codium.functions.${system}) writeSettingsJSON mkCodium;
+      inherit (my-codium.configs.${system}) extensions settingsNix;
+      inherit (devshell.functions.${system}) mkCommands mkShell;
+      inherit (haskell-tools.functions.${system}) haskellTools;
+      inherit (drv-tools.functions.${system}) mkBin mkShellApp;
+      lima = my-lima.packages.${system}.default;
+
+      # Next, set the desired GHC version
+      ghcVersion_ = "92";
+
+      # and the name of the package
+      myPackageName = "fused-effects";
+
+      # Here's our override
+      override = {
+        overrides = self: super: {
+          myPackage = super.callCabal2nix myPackageName ./. { };
+        };
+      };
+
+      # We supply it to a helper function that will give us haskell tools for a given 
+      # compiler version, override, packages we're going to develop, and their binary runtime dependencies
+
+      inherit (haskellTools ghcVersion_ override (ps: [ ps.myPackage ]) [ ]) hls cabal;
+
+      codiumTools = [ cabal hls ];
+
+      # We compose VSCodium with dev tools and HLS
+      codium = mkCodium {
+        extensions = { inherit (extensions) nix haskell misc github markdown; };
+        runtimeDependencies = codiumTools;
+      };
+
+      # Write .vscode/settings.json
+      writeSettings = writeSettingsJSON {
+        inherit (settingsNix) haskell todo-tree files editor gitlens
+          git nix-ide workbench markdown-all-in-one markdown-language-features;
+      };
```

Comment:
As mentioned above, I don’t think our project is the right place to put editor-specific settings, so I’d prefer to see this elided.